tests/test_shira.py

```diff
@@ -0,0 +1,328 @@
+
+        manual_merge_weight1 = peft_model.base_model.model.lin1.base_layer.weight + peft_model.base_model.model.lin1.get_delta_weight('default')
+        manual_merge_weight2 = peft_model.base_model.model.lin2.base_layer.weight + peft_model.base_model.model.lin2.get_delta_weight('default')
+
+        peft_model.base_model.model.lin1.merge(adapter_names=['default'])
+        peft_model.base_model.model.lin2.merge(adapter_names=['default'])
+
+        assert torch.all(manual_merge_weight1 - peft_model.base_model.model.lin1.base_layer.weight.data == 0)
+        assert torch.all(manual_merge_weight2 - peft_model.base_model.model.lin2.base_layer.weight.data == 0)
+
+        manual_unmerge_weight1 = peft_model.base_model.model.lin1.base_layer.weight - peft_model.base_model.model.lin1.get_delta_weight('default')
+        manual_unmerge_weight2 = peft_model.base_model.model.lin2.base_layer.weight - peft_model.base_model.model.lin2.get_delta_weight('default')
+
+        peft_model.base_model.model.lin1.unmerge()
+        peft_model.base_model.model.lin2.unmerge()
+
+        assert torch.all(peft_model.base_model.model.lin1.base_layer.weight.data - manual_unmerge_weight1 == 0)
+        assert torch.all(peft_model.base_model.model.lin2.base_layer.weight.data - manual_unmerge_weight2 == 0)
+
+        peft_model.merge_and_unload()
+
+        assert torch.all(peft_model.base_model.model.lin1.weight.data - manual_merge_weight1 == 0)
+        assert torch.all(peft_model.base_model.model.lin2.weight.data - manual_merge_weight2 == 0)
+
+        return peft_model
+
+    def test_mlp_multiple_adapters_same_attachment_points(self, mlp):
+        # torch.manual_seed(0)
+
+        config = ShiraConfig(r=2, target_modules=["lin1", "lin2"])
+        # creates a default SHiRA adapter
+        peft_model = get_peft_model(mlp, config)
+        config2 = ShiraConfig(r=2, target_modules=["lin1", "lin2"])
+        peft_model.add_adapter("other", config2)
+
+        assert len(peft_model.base_model.model.lin1.shira_weight.keys()) == 2
+        assert len(peft_model.base_model.model.lin2.shira_weight.keys()) == 2
+
+        return peft_model
+
+    def test_mlp_multiple_adapters_different_attachment_points(self, mlp):
+        # torch.manual_seed(0)
+
+        config = ShiraConfig(r=2, target_modules=["lin1", "lin2"])
+        # creates a default SHiRA adapter
+        peft_model = get_peft_model(mlp, config)
+        config2 = ShiraConfig(r=2, target_modules=["lin1", "lin2", "lin3"], random_seed=52)
+        peft_model.add_adapter("other", config2)
+
+        assert len(peft_model.base_model.model.lin1.shira_weight.keys()) == 2
+        assert len(peft_model.base_model.model.lin2.shira_weight.keys()) == 2
+        assert len(peft_model.base_model.model.lin3.shira_weight.keys()) == 1
+
+        return peft_model
+
+    def test_mlp_multiple_adapters_different_attachment_points_merge_and_unload(self, mlp):
```

Comment:
This will be covered by `test_custom_models.py` if you add entries here: https://github.com/huggingface/peft/blob/6bcefb02c66ff6821fbbe58bc4d8fc864e2db335/tests/test_custom_models.py#L549.